### PR TITLE
feat: Add notes functionality

### PR DESCRIPTION
**Pull request by Hardiegogo**
Added add notes functionality:
- created a custom hook (useNotes) which internally uses NotesContext.
- added css for desktop version

### CODE REVIEW

**Review by AmishMishra11 — APPROVED** [review]
Clean code
Liked how you have checked for auth before sending user to any page

File: src/components/new-note-component/NewNote.jsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import React, { useState } from "react";
+import { addNote } from "../../utils/notes-util/addNote";
+import { useNotes } from "../../context/useNotes";
+import "./NewNote.css";
+
+const NewNote = ({createNoteState:{createNote,setCreateNote}}) => {
+  const [tempNote, setTempNote] = useState({
+    _id: "",
```
Comment (VedanthB) [inline src/components/new-note-component/NewNote.jsx]: just a suggestion, make a variable for `initNoteState = { ... }`, then pass that into the initialState, makes it more readable

File: src/context/useNotes.jsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import { createContext,useContext,useReducer } from "react";
+
+
+const NotesContext=createContext(null)
+
+export const useNotes=()=>useContext(NotesContext)
+
+const notesReducer=(state,action)=>{
```
Comment (VedanthB) [inline src/context/useNotes.jsx]: you should move your reducers into a diff folder

File: src/index.js [hunk 3]
```diff
@@ -5,17 +5,19 @@ import App from "./App";
 import { makeServer } from "./server";
 import { BrowserRouter } from "react-router-dom";
 import { AuthProvider } from "./context/useAuth";
```
Comment (VedanthB) [inline src/index.js:7]: you can do a barrel import/export for everything, makes the code more clean

**Review by VedanthB — APPROVED** [review]
Good work Chirag

 things i liked
- clean organised code
- good variable naming
- consistent spacing units 

You can merge the PR

File: src/context/useNotes.jsx (hunk 2 above)
Comment (Hardiegogo) [reply src/context/useNotes.jsx]: Got it.

File: src/components/new-note-component/NewNote.jsx (hunk 1 above)
Comment (Hardiegogo) [reply src/components/new-note-component/NewNote.jsx]: Yes I will make this change

**Pull request merged**